src/Core/System/SalesChannel/Context/SalesChannelContextService.php

```diff
@@ -116,6 +116,10 @@ public function get(SalesChannelContextServiceParameters $parameters): SalesChan
                 $this->cartService->setCart($result->getCart());
             }
 
+            if ($esiRequest) {
```

Comment:
```suggestion
            } else {
```

shouldn't this be done on every else case? :thinking: 

@cyl3x would have so say something about that cart changes, but in general this looks okay for me :+1:

Reply:
This would also include the main request if the cart is already loaded, which would be unnecessary (assuming that the context is correctly hydrated) 🤔